### PR TITLE
Feature/resolve alias

### PR DESCRIPTION
**Pull request by justame**
@yanivefraim @shlomitc

### CODE REVIEW

File: config/project.js [hunk 1]
```diff
@@ -35,7 +35,8 @@ module.exports = {
   separateCss: () => getConfig('separateCss', true),
   cssModules: () => getConfig('cssModules', true),
   externals: () => getConfig('externals'),
-  babel: () => _.get(packagejson, 'babel')
+  babel: () => _.get(packagejson, 'babel'),
+  getPathsAlias: () => getConfig('pathsAlias', {})
```
Comment (mastertheblaster) [inline config/project.js]: The config property doesn't sound properly:
`aliasedPaths`
`pathAliases`
`importAliases`
...

File: config/webpack.config.client.js [hunk 2]
```diff
@@ -31,11 +31,7 @@ const config = ({debug, hot} = {}) => {
       ...extractCSS ? [extractCSS] : [],
 
       ...debug ? [] : [
-        new webpack.optimize.UglifyJsPlugin({
```
Comment (mastertheblaster) [inline config/webpack.config.client.js]: Is this really ok ? We wont our javascript to be minified.

File: test/webpack-config.spec.js [hunk 3]
```diff
@@ -87,5 +90,20 @@ describe('Webpack basic configs', () => {
...
+    it('should resolve paths imported from package.json', () => {
+      test.setup({
+        'package.json': fx.packageJson({
+          pathsAlias: {submodule: 'src/components/modules/submodule'}
+        }, {
+          'babel-preset-es2015': 'latest'
+        }),
+        '.babelrc': `{"presets": ["es2015"]}`,
+        'components/modules/submodule/index.js': 'export const component = 123;',
+        'client.js': `import {component} from 'submodule';`
```
Comment (mastertheblaster) [inline test/webpack-config.spec.js]: It should work with `require('submodule')` as well?

**Comment by yanivefraim** [comment]
@justame - build is failing (the new test is red). Can you plz check this + we should discuss how to fix the minify/uglify issue

File: config/webpack.config.client.js (hunk 2 above)
Comment (roymiloh) [inline config/webpack.config.client.js]: No

File: config/project.js (hunk 1 above)
Comment (justame) [inline config/project.js]: no problem, I'll change it

**Comment by shahata** [comment]
I agree with @tqmukas, this is a partial solution since it doesn’t work on path’s, plus I also feel it is not the best solution for the problem. there’s a very long discussion about this issue in https://gist.github.com/branneman/8048520 we should look more deeply in to it and find the best solution. Sorry @mastertheblaster @justame, I’m reverting this commit and we’ll discuss soon.

**Pull request merged**